### ISSUE TITLE
Cannot push to my repo (mirror is readonly)

### ISSUE THREAD

**Issue by clementrey** [go-gitea/gitea#7609]
- Gitea version (or commit ref): 1.5
- Git version: 2.17.1
- Operating system: Debian buster
- Database (use `[x]`): 
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:

## Description

Hello

I have a repository hosted by gitea running locally on my machine (through a VM). This repository is mirroring a remote repositry hosted somewhere else.
When I try to push to my gitea repository, I have the following error message (I have added verbose flags)

```
remote: mirror repository is read-only
fatal: unable to access 'http://X.X.X.X:YYYYY/myUser/myPorject.git/': The requested URL returned error: 403

```
I don't know why mirror repository is in Read Only. I have checked and I can very well pushed commit to the mirrored repo.

Do you what is the reason of this error ?
Best Regards

**Comment by zeripath** [comment]
You cannot push to mirrored repositories. You will either need to fork the mirror and update the fork from the mirror intermittently manually, or, you need to push to the upstream.

If we allow pushes to mirrors and a conflicting change from upstream occurs what is supposed to happen?

**Comment by SoulSeekkor** [comment]
Running into this same problem, could do this in Bitbucket by forking a repo and telling it to keep it in sync.  This way I was able to create new branches for what I wanted to change, this worked perfectly as the forked repo would pull in new tags and commits on master, etc. and as long as I didn't do something silly like name a branch something that would be synced it's fine doing so (last time when I commited to master to see what would happen it simply stops syncing as the history has changed).

This is built-in functionality in Bitbucket.  I've included a screenshot of what this looks like after you click create fork next to a repo in that, it would be great to see this in Gitea as well.
![Untitled](https://user-images.githubusercontent.com/16005426/62153310-e748f680-b2c9-11e9-9541-e141a7bddc22.png)

**Comment by lunny** [comment]
You could create a mirror of that repository.

**Comment by SoulSeekkor** [comment]
THat doesn't work as the mirror is readonly.

**Comment by SoulSeekkor** [comment]
If nothing else in my case it'd be great if on the fork of the repo I could click a button to update a specific branch to the latest or something similar.

**Comment by KenMercusLai** [comment]
you can convert it to a regular repo in the setting page of your repo.

**Comment by EslamFouadd** [comment]
The GCP repositories shell still having the same problem when I commit and push a change to the files in my repos on GCP.

Any updates for the solutions?

**Comment by qiangzai201611** [comment]
You need to set the image warehouse to a regular warehouse, otherwise prompt: remote: mirror repository is read-only

**Comment by qiangzai201611** [comment]
![image](https://user-images.githubusercontent.com/24547867/225250280-87d143bd-4b6f-420c-82a2-c30dc0973e01.png)

**Comment by hermannoffen** [comment]
If it shouldn't be possible for a mirror repository to pull and push from/to remotes, then why is it configurable like this?
I cloned a GitHub organisation repository as a pull mirror and added it as a push target afterwards as well. Synchronizing the push mirror manually from gitea's repo settings was successful, then. But pushing to my gitea repo ended up with the error message mentioned in this issue.

**Comment by letto4135** [comment]
I agree with @hermannoffen, if you can't set up push pull syncing it shouldn't be an option, and since you can set that up the mirror shouldn't be read only.

**Comment by rushimahindrakar** [comment]
I am also facing the same issue. is there any option now to without create new repo?

**Comment by lunny** [comment]
Maybe we could have a branch list configuration that will be mirrored and then other branches could be pushed. The default value is all branches are mirrored but allow users to change.

**Comment by lunny** [comment]
> If nothing else in my case it'd be great if on the fork of the repo I could click a button to update a specific branch to the latest or something similar.

This has been implemented in v1.23.1
